modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexPlugin.java

```diff
@@ -40,7 +41,8 @@
         return Arrays.asList(new ActionHandler<>(ReindexAction.INSTANCE, TransportReindexAction.class),
                 new ActionHandler<>(UpdateByQueryAction.INSTANCE, TransportUpdateByQueryAction.class),
                 new ActionHandler<>(DeleteByQueryAction.INSTANCE, TransportDeleteByQueryAction.class),
-                new ActionHandler<>(RethrottleAction.INSTANCE, TransportRethrottleAction.class));
+                new ActionHandler<>(RethrottleAction.INSTANCE, TransportRethrottleAction.class),
+                new ActionHandler<>(MigrateIndexAction.INSTANCE, TransportMigrateIndexAction.class));
```

Comment:
Keeping this transport action in the reindex module while the request and response are in core sucks. We're not really sure what the right way to fix it is because we want other Elasticsearch plugins to be able to use this request and if it is in the reindex plugin this is hard. We want it in the reindex module because it relies on reindex and we don't want to yank reindex into core.